tachyon/zk/plonk/expressions/evaluator/identifier.h

```diff
@@ -20,71 +20,83 @@
 namespace tachyon::zk::plonk {
 
 template <typename F>
-std::string GetIdentifier(const Expression<F>* input) {
-  std::ostringstream ss;
-  switch (input->type()) {
-    case ExpressionType::kConstant:
-      ss << input->ToConstant()->value().ToString();
-      return ss.str();
-    case ExpressionType::kNegated:
-      ss << "(-";
-      ss << GetIdentifier(input->ToNegated()->expr());
-      ss << ")";
-      return ss.str();
-    case ExpressionType::kSum:
-      ss << "(";
-      ss << GetIdentifier(input->ToSum()->left());
-      ss << "+";
-      ss << GetIdentifier(input->ToSum()->right());
-      ss << ")";
-      return ss.str();
-    case ExpressionType::kProduct:
-      ss << "(";
-      ss << GetIdentifier(input->ToProduct()->left());
-      ss << "*";
-      ss << GetIdentifier(input->ToProduct()->right());
-      ss << ")";
-      return ss.str();
-    case ExpressionType::kScaled:
-      ss << "*" << input->ToScaled()->scale().ToString();
-      return ss.str();
-    case ExpressionType::kFixed: {
-      const FixedExpression<F>* fixed = input->ToFixed();
-      ss << "fixed[" << fixed->query().column().index() << "]["
-         << fixed->query().rotation().value() << "]";
-      return ss.str();
-    }
-    case ExpressionType::kAdvice: {
-      const AdviceExpression<F>* advice = input->ToAdvice();
-      ss << "advice[" << advice->query().column().index() << "]["
-         << advice->query().rotation().value() << "]";
-      return ss.str();
-    }
-    case ExpressionType::kInstance: {
-      const InstanceExpression<F>* instance = input->ToInstance();
-      ss << "instance[" << instance->query().column().index() << "]["
-         << instance->query().rotation().value() << "]";
-      return ss.str();
-    }
-    case ExpressionType::kChallenge: {
-      const ChallengeExpression<F>* challenge = input->ToChallenge();
-      ss << "challenge[" << challenge->challenge().index() << "]";
-      return ss.str();
-    }
-    case ExpressionType::kSelector: {
-      const SelectorExpression<F>* selector = input->ToSelector();
-      ss << "selector[" << selector->selector().index() << "]";
-      return ss.str();
+class Identifier : public Evaluator<F, std::string> {
+ public:
+  // Evaluator methods
+  std::string Evaluate(const Expression<F>* input) override {
+    switch (input->type()) {
+      case ExpressionType::kConstant:
+        ss_ << input->ToConstant()->value().ToString();
+        return ss_.str();
+      case ExpressionType::kNegated:
+        ss_ << "(-";
+        ss << Evaluate(input->ToNegated()->expr());
```

Comment:
Please fix this commit and ensure it works by changing instances of `ss` to `ss_`.